src/Microsoft.DotNet.Interactive.SqlServer/MsSqlKernel.cs

```diff
@@ -129,25 +129,31 @@ public async Task HandleAsync(SubmitCode command, KernelInvocationContext contex
                                 return;
                             }
 
-                            var subsetParams = new QueryExecuteSubsetParams()
+                            CellValue[][] rows = null;
+                            if (resultSummary.RowCount > 0)
                             {
-                                OwnerUri = _tempFileUri.ToString(),
-                                BatchIndex = batchSummary.Id,
-                                ResultSetIndex = resultSummary.Id,
-                                RowsStartIndex = 0,
-                                RowsCount = Convert.ToInt32(resultSummary.RowCount)
-                            };
-                            var subsetResult = await _serviceClient.ExecuteQueryExecuteSubsetAsync(subsetParams);
-
-                            if (subsetResult.Message != null)
-                            {
-                                context.Fail(message: subsetResult.Message);
-                            }
-                            else
-                            {
-                                var table = GetEnumerableTable(resultSummary.ColumnInfo, subsetResult.ResultSubset.Rows);
-                                context.Display(table);
+                                var subsetParams = new QueryExecuteSubsetParams()
+                                {
+                                    OwnerUri = _tempFileUri.ToString(),
+                                    BatchIndex = batchSummary.Id,
+                                    ResultSetIndex = resultSummary.Id,
+                                    RowsStartIndex = 0,
+                                    RowsCount = Convert.ToInt32(resultSummary.RowCount)
+                                };
+                                var subsetResult = await _serviceClient.ExecuteQueryExecuteSubsetAsync(subsetParams);
+                                if (subsetResult.Message != null)
+                                {
+                                    context.Fail(message: subsetResult.Message);
```

Comment:
Is a message always an error?

Reply:
It used to be, but now it looks like the property was removed on the SqlToolsService side: https://github.com/microsoft/sqltoolsservice/blob/main/src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Contracts/SubsetRequest.cs#L47

I've removed it from our subset result class.